scripts/qa_crowbarsetup.sh

```diff
@@ -5038,10 +5039,147 @@ function ping_fips
     done
 }
 
+function get_stack_fips
+{
+    local stack=$1
+    local fip_file=$2
+    local all_fips=$(mktemp)
+    local fip_uuids=$(openstack stack resource list $stack -f value -c physical_resource_id --filter type=OS::Neutron::FloatingIP)
+    openstack floating ip list -f value > $all_fips
```

Comment:
`echo "$fip_uuids" | xargs -i openstack floating ip show -f value -c floating_ip_address {}` should give you the same result as this call and the loop-with-awk below.

Reply:
if you want to save some `openstack` calls you can also do:
```
local fips=$(openstack floating ip list -f value -c ID -c "Floating IP Address" )
echo "$fip_uuids" | xargs -i sh -c "echo \"$fips\" | grep {}" | cut -d' ' -f2

Reply:
...buffering things in memory like that requires a lot of ugly and brittle escaping and extra processing. That makes the code less readable and more error prone.

Reply:
even with file I'd still do something like this instead of the awk loop:
```
echo "$fip_uuids" | xargs -i sh -c "grep {} $all_fips" | cut -d' ' -f... # field depending on the format of all_fips file
```

Reply:
Uhm. That is not more readable though. And you've got a pipeline of 3 processes (plus a 4th for the shell `xargs` spawns) instead of just the one `awk` process.